static/app/components/notificationActions/forms/pagerdutyForm.tsx

```diff
@@ -0,0 +1,146 @@
+import {useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+import {
+  NotificationActionCell,
+  NotificationActionFormContainer,
+} from 'sentry/components/notificationActions/notificationActionItem';
+import {IconChevron} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+} from 'sentry/types/notificationActions';
+
+interface PagerdutyFormProps {
+  /**
+   * The notification action being represented
+   */
+  action: Partial<NotificationAction>;
+  onCancel: () => void;
+  onChange: (name: string, value: any) => void;
+  onSave: () => void;
+  /**
+   * Map of pagerduty integration IDs to available actions for those IDs
+   */
+  pagerdutyIntegrations: Record<number, AvailableNotificationAction[]>;
+}
+
+function PagerdutyForm({
+  action,
+  onCancel,
+  onChange,
+  onSave,
+  pagerdutyIntegrations,
+}: PagerdutyFormProps) {
+  const [selectedAccount, setSelectedAccount] = useState(
+    action.integrationId
+      ? pagerdutyIntegrations[action.integrationId][0].action.integrationName
+      : ''
+  );
+  const [selectedDisplay, setSelectedDisplay] = useState(action.targetDisplay ?? '');
+
+  const accountOptions = Object.keys(pagerdutyIntegrations).map<MenuItemProps>(
+    integrationId => {
+      // Get the name of the integration for the integrationId from the first
+      // AvailableNotificationAction element in the array
+      const integrationName =
+        pagerdutyIntegrations[integrationId][0].action.integrationName;
+      return {
+        key: integrationName,
+        label: integrationName,
+        onAction: value => {
+          onChange('integrationId', integrationId);
+          setSelectedAccount(value);
+        },
+      };
+    }
+  );
+
+  // Each Pagerduty account has its own list of services
+  function getServiceOptions(): MenuItemProps[] {
+    if (!action.integrationId) {
+      return [];
+    }
+    const services = pagerdutyIntegrations[action.integrationId];
+    const serviceOptions = services.map<MenuItemProps>(service => ({
+      key: service.action.targetDisplay ?? '',
+      label: service.action.targetDisplay,
+      onAction: value => {
+        onChange('targetDisplay', service.action.targetDisplay);
+        setSelectedDisplay(value);
+        onChange('targetIdentifier', service.action.targetIdentifier);
```

Comment:
nit: My smooth brain would like if the two onChange calls are beside one another 🧠